src/config.js

```diff
@@ -0,0 +1,56 @@
+/* @flow */
+import requireUncached from 'require-uncached';
+
+import {createLogger} from './util/logger';
+import {UsageError} from './errors';
+
+const log = createLogger(__filename);
+
+type ApplyConfigToArgvParams = {|
+  argv: Object,
+  configObject: Object,
+  defaultValues: Object,
+|};
+
+export function applyConfigToArgv({
+  argv,
+  configObject,
+  defaultValues = {},
+}: ApplyConfigToArgvParams): Object {
+  const newArgv = {...argv};
+  for (const option in configObject) {
+    const wasValueSetOnCLI = typeof(newArgv[option]) !== 'undefined' &&
+      (argv[option] !== defaultValues[option]);
+    if (wasValueSetOnCLI) {
+      log.debug(`Favoring CLI: ${option}=${argv[option]} over ` +
+        `configuration: ${option}=${configObject[option]}`);
+      continue;
+    }
+    if (!newArgv.hasOwnProperty(option) || defaultValues[option] ||
```

Comment:
There are a couple things I'd like to see changed here:

* Check for an `undefined` value instead of using `hasOwnProperty` because that will work even in a future case where yargs uses prototype inheritance to build up an `argv` object (it's unlikely but possible)
* Remove the second part of the if statement because it's redundant now that your `wasValueSetOnCLI` statement is working

Here is a suggested patch:
````diff
diff --git a/src/config.js b/src/config.js
index 3735ae6..4a92a7e 100644
--- a/src/config.js
+++ b/src/config.js
@@ -26,14 +26,15 @@ export function applyConfigToArgv({
         `configuration: ${option}=${configObject[option]}`);
       continue;
     }
-    if (!newArgv.hasOwnProperty(option) || defaultValues[option] ||
-      typeof(newArgv[option]) === 'boolean' ||
-      typeof(newArgv[option]) === 'number') {
-
-      newArgv[option] = configObject[option];
-      log.debug(`Favoring configuration: ${option}=${configObject[option]} ` +
-        `over CLI: ${option}=${argv[option]}`);
+    if (typeof(argv[option]) === 'undefined') {
+      log.debug(`Ignoring config option: ${option}=${configObject[option]} ` +
+        'because this is an unknown option');
+      continue;
     }
+
+    newArgv[option] = configObject[option];
+    log.debug(`Favoring configuration: ${option}=${configObject[option]} ` +
+      `over CLI: ${option}=${argv[option]}`);
   }
   return newArgv;
 }
@@ -53,4 +54,4 @@ export function loadJSConfigFile(filePath: string): Object {
       'Did you set module.exports = {...}?');
   }
   return configObject;
-}
\ No newline at end of file
+}

````